File: AgoraChat-ObjectiveC/AgoraChat/Code/Utilities/UITabBar+RedPoint.m
//
//  UITabBar+RedPoint.m
//  ChatDemo-UI3.0
//
//  Created by liang on 2021/11/18.
//  Copyright © 2021 easemob. All rights reserved.
//

#import "UITabBar+RedPoint.h"
#define TabbarItemNums 3.0

@implementation UITabBar (RedPoint)

- (void)showBadgeOnItemIndex:(int)index{
    [self removeBadgeOnItemIndex:index];
   
    UIView *badgeView = [[UIView alloc]init];
    badgeView.tag = 888 + index;
    badgeView.layer.cornerRadius = 5;
    badgeView.backgroundColor = TextLabelPinkColor;
    CGRect tabFrame = self.frame;
   
    float percentX = (index +0.6) / TabbarItemNums;
    CGFloat x = ceilf(percentX * tabFrame.size.width);
    CGFloat y = ceilf(0.1 * tabFrame.size.height);
    badgeView.frame = CGRectMake(x, y, 10, 10);
    [self addSubview:badgeView];
}

- (void)hideBadgeOnItemIndex:(int)index{
    [self removeBadgeOnItemIndex:index];
}

- (void)removeBadgeOnItemIndex:(int)index{
    for (UIView *subView in self.subviews) {
        if (subView.tag == 888+index) {
            [subView removeFromSuperview];
        }
    }
}

@end

#undef TabbarItemNums

File: AgoraChat-Swift/AgoraChat-Swift/code/Utilities/UITabBar+RedPoint.swift
//
//  UITabBar+RedPoint.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/12.
//

import UIKit

extension UITabBar {
    func showBadge(index: Int) {
        self.removeBadge(index: index)
        
        let badgeView = UIView()
        badgeView.tag = 888 + index
        badgeView.layer.cornerRadius = 5
        badgeView.backgroundColor = Color_FF14CC
        
        let tabFrame = self.frame
        let percentX = (CGFloat(index) + 0.6) / 3.0
        let x = ceil(percentX * tabFrame.width)
        let y = ceil(0.1 * tabFrame.height)
        badgeView.frame = CGRect(x: x, y: y, width: 10, height: 10)
        self.addSubview(badgeView)
    }
    
    func hideBadge(index: Int) {
        removeBadge(index: index)
    }
    
    private func removeBadge(index: Int) {
        for view in self.subviews where view.tag == 888 + index {
            view.removeFromSuperview()
            return
        }
    }
}
